BaseClasses.py

```diff
@@ -571,10 +570,10 @@ def sweep_for_events(self, key_only=False, locations=None):
 
 
     def can_reach_blue(self, region, player):
-        return region in self.reachable_regions[player] and self.reachable_regions[player][region] in [CrystalBarrier.Blue, CrystalBarrier.Either]
+        return region in self.reachable_regions[player] and self.reachable_regions[player][region] in [CrystalBarrier.Blue, CrystalBarrier.Either] and self.can_hit_crystal(player)
```

Comment:
I feel like this is the wrong place to check if you can hit a switch.  Some rooms have convenient pots you can use to hit the switch with no equipment.

Reply:
I agree can-hit-switch should be checked somewhere else. This started as a placeholder for me when working out bomb-logic. This check should probably be worked out for each switch separately instead.

Reply:
The way this is complicated to accommodate is that during dungeon transversal crystal-state is changed by going through doors in regions with crystal switches. To differentiate between crystals that can be hit with a pot and crystals that need a weapon, I would have to expand the current work quite a bit. I just dont don't think there is much to gain from that right now to warrant that time. So better to safe with a can_hit_switch so that pot-throwing becomes a sequence break the few times it comes up. 

That said, it wouldn't be a problematic to add, and I'd do it there's more reasons to work that out?

Reply:
I agree, this is good for a first pass at the logic.